File: Domain/Entities/Task.cs
using Domain.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Entities
{
    public class Task
    {
        [Key]
        public Guid Id { get; set; }
        public string Description { get; set; }
        public bool IsComplete { get; set; }
        public TaskPriority Priority { get; set; }
        public TaskType Type { get; set; }

    }
}

File: client-app/src/models/Task.ts
export interface Task {
    id : string;
    description : string;
    isComplete : boolean;
    priority : number;
    type : number;
}

export class Task implements Task {

}

export class PostTask {
    description : string;
    priority : number;
    type : number;

    constructor(description: string , priority: number , type: number){
        this.description = description;
        this.priority = priority;
        this.type = type;
    }
}